postgres/client.sls

```diff
@@ -1,17 +1,18 @@
-{% from "postgres/map.jinja" import postgres with context %}
+{%- from "postgres/map.jinja" import postgres with context -%}
+
+{%- set pkgs = [] %}
+{%- for pkg in (postgres.pkg_client, postgres.pkg_libpq_dev) %}
+  {%- if pkg %}
+    {%- do pkgs.append(pkg) %}
+  {%- endif %}
+{%- endfor -%}
 
-{% if postgres.use_upstream_repo %}
```

Comment:
In general, my PR is all about to make `upstream.sls` *stateful*.

And it makes sense when you *remove* upstream repository configuration.